src/s390_ecc.c

```diff
@@ -859,6 +867,9 @@ unsigned int ecdh_sw(const ICA_EC_KEY *privkey_A, const ICA_EC_KEY *pubkey_B,
 	if (rc == 0)
 		goto err;
 
+	if (!EC_KEY_check_key(a) || !EC_KEY_check_key(b))
```

Comment:
I think this is superfluous here. You get ta and b from make_eckey() and make_public_eckey() above, and both do the check already. So no need to check them again here.

Reply:
Yes that's true. Removed.